common/types.go

```diff
@@ -22,35 +22,48 @@ package common
 
 import (
 	"bytes"
+	"encoding/binary"
 	"encoding/hex"
-	"errors"
 	"fmt"
 	"math/big"
 	"math/rand"
 	"reflect"
+	"sync/atomic"
+	"time"
 
 	"github.com/klaytn/klaytn/common/hexutil"
 	"github.com/klaytn/klaytn/crypto/sha3"
 )
 
 const (
-	HashLength      = 32
-	AddressLength   = 20
-	SignatureLength = 65
+	HashLength         = 32
+	ExtHashNonceLength = 7
+	ExtHashLength      = HashLength + ExtHashNonceLength
+	AddressLength      = 20
+	SignatureLength    = 65
 )
 
 var (
 	hashT    = reflect.TypeOf(Hash{})
+	extHashT = reflect.TypeOf(ExtHash{})
 	addressT = reflect.TypeOf(Address{})
 )
 
-var lastPrecompiledContractAddressHex = hexutil.MustDecode("0x00000000000000000000000000000000000003FF")
-
 var (
-	errStringLengthExceedsAddressLength = errors.New("the string length exceeds the address length (20)")
-	errEmptyString                      = errors.New("empty string")
+	lastPrecompiledContractAddressHex = hexutil.MustDecode("0x00000000000000000000000000000000000003FF")
+
+	extHashCounterMin  = uint64(0x100) // Any counter below extHashCounterMin are reserved.
+	extHashCounter     = extHashCounterMin
+	extHashLegacyNonce = ExtHashNonce{0, 0, 0, 0, 0, 0, 0x45}
```

Comment:
The value doesn't have any meaning, right? Please comment about this for the later changes

Reply:
@ethan-kr Can you explain the meaning of the value? How about changing this to all-zero?

Reply:
Added comments as to why we have nonzero number.